### PR TITLE
Implement storage capacity

### PR DESCRIPTION
**Pull request by SeptimiuVana**

### CODE REVIEW

File: apps/uvc-app/uvc-gadget.cpp [hunk 1]
```diff
@@ -455,6 +455,22 @@ uvc_payload::ServerResponse handleClientRequest(const uvc_payload::ClientRequest
...
+  case uvc_payload::FunctionName::STORAGE_READ_CAPACITY: {
+
+    size_t *data = new size_t;
+
+    aditof::Status status = storages[0]->getCapacity(data);
+    LOG(INFO) << " READ SIZE " << *data;
+    if (status == aditof::Status::OK) {
+      response.add_bytes_payload(data,sizeof(size_t));
+    }
+    delete[] data;
```
Comment (dNechita) [inline apps/uvc-app/uvc-gadget.cpp]: delete with [] is used when multiple elements are allocated with new.  Also you can simply declare it as a local variable: size_t data

Comment (dNechita) [reply apps/uvc-app/uvc-gadget.cpp]: Also, getCapacity expects a reference. You are providing a pointer.

File: apps/uvc-app/uvc-gadget.cpp [hunk 2]
```diff
@@ -455,6 +455,22 @@ uvc_payload::ServerResponse handleClientRequest(const uvc_payload::ClientRequest
...
   }
 
+  case uvc_payload::FunctionName::STORAGE_READ_CAPACITY: {
+
+    size_t *data = new size_t;
+
+    aditof::Status status = storages[0]->getCapacity(data);
+    LOG(INFO) << " READ SIZE " << *data;
+    if (status == aditof::Status::OK) {
+      response.add_bytes_payload(data,sizeof(size_t));
```
Comment (dNechita) [inline apps/uvc-app/uvc-gadget.cpp]: You can use int32_payload instead of bytes_payload

File: apps/uvc-app/uvc-gadget.cpp (hunk 1 above)
Comment (SeptimiuVana) [reply apps/uvc-app/uvc-gadget.cpp]: Was fixed in second commit.

File: sdk/src/connections/usb/windows/usb_storage_windows.cpp [hunk 3]
```diff
@@ -221,8 +221,55 @@ Status UsbStorage::write(const uint32_t address, const uint8_t *data,
...
+    }
+
+    DLOG(INFO) << "Received the following message: " << responseMsg.DebugString();
+
+    if (responseMsg.status() != usb_payload::Status::OK) {
+        LOG(ERROR) << "Read memory size operation failed on UVC gadget";
+        return static_cast<aditof::Status>(responseMsg.status());
+    }
+   
+    memcpy(&nbBytes, responseMsg.bytes_payload(0).c_str(), responseMsg.bytes_payload(0).length());
```
Comment (dNechita) [inline sdk/src/connections/usb/windows/usb_storage_windows.cpp:270]: you won't be needing memcpy if you use int32_payload

File: apps/uvc-app/uvc-gadget.cpp (hunk 1 above)
Comment (dNechita) [reply apps/uvc-app/uvc-gadget.cpp]: oh, ok. I was reviewing one commit at a time

File: sdk/src/connections/usb/windows/usb_storage_windows.cpp (hunk 3 above)
Comment (SeptimiuVana) [reply sdk/src/connections/usb/windows/usb_storage_windows.cpp]: Fixed.

File: apps/uvc-app/uvc-gadget.cpp (hunk 2 above)
Comment (SeptimiuVana) [reply apps/uvc-app/uvc-gadget.cpp]: Fixed.

**Review by dNechita — APPROVED** [review]

**Pull request merged**